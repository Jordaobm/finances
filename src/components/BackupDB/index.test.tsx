import { NavigationContainer } from "@react-navigation/native";
import React from "react";
import { BackupDB } from ".";
import renderer, { act } from "react-test-renderer";

import { render, fireEvent } from "@testing-library/react-native";

test("Component BackupDB load data", async () => {
  let loadingDataAPP = 0;

  function setLoadingDataAPP(value: number) {
    loadingDataAPP = value;
  }

  let loadingDataDB = 0;

  function setLoadingDataDB(value: number) {
    loadingDataDB = value;
  }

  function setBackupDB(value: boolean) {
    return value;
  }

  const tree = renderer
    .create(
      <NavigationContainer>
        <BackupDB
          loadingDataAPP={loadingDataDB}
          loadingDataDB={loadingDataDB}
          setBackupDB={setBackupDB}
          setStatusLoadingDataForApp={setLoadingDataAPP}
          setStatusLoadingDataForDB={setLoadingDataDB}
        />
      </NavigationContainer>
    )
    .toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});

test("Component BackupDB load APP complete", async () => {
  let loadingDataAPP = 100;

  function setLoadingDataAPP(value: number) {
    loadingDataAPP = value;
  }

  let loadingDataDB = 10;

  function setLoadingDataDB(value: number) {
    loadingDataDB = value;
  }

  function setBackupDB(value: boolean) {
    return value;
  }

  const tree = renderer
    .create(
      <NavigationContainer>
        <BackupDB
          loadingDataAPP={loadingDataAPP}
          loadingDataDB={loadingDataDB}
          setBackupDB={setBackupDB}
          setStatusLoadingDataForApp={setLoadingDataAPP}
          setStatusLoadingDataForDB={setLoadingDataDB}
        />
      </NavigationContainer>
    )
    .toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});

test("Component BackupDB load DB complete", async () => {
  let loadingDataAPP = 100;

  function setLoadingDataAPP(value: number) {
    loadingDataAPP = value;
  }

  let loadingDataDB = 100;

  function setLoadingDataDB(value: number) {
    loadingDataDB = value;
  }

  function setBackupDB(value: boolean) {
    return value;
  }

  const tree = renderer
    .create(
      <NavigationContainer>
        <BackupDB
          loadingDataAPP={loadingDataAPP}
          loadingDataDB={loadingDataDB}
          setBackupDB={setBackupDB}
          setStatusLoadingDataForApp={setLoadingDataAPP}
          setStatusLoadingDataForDB={setLoadingDataDB}
        />
      </NavigationContainer>
    )
    .toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});

test("Component BackupDB load DB complete", async () => {
  let loadingDataAPP = 90;

  function setLoadingDataAPP(value: number) {
    loadingDataAPP = value;
  }

  let loadingDataDB = 100;

  function setLoadingDataDB(value: number) {
    loadingDataDB = value;
  }

  function setBackupDB(value: boolean) {
    return value;
  }

  const tree = renderer
    .create(
      <NavigationContainer>
        <BackupDB
          loadingDataAPP={loadingDataAPP}
          loadingDataDB={loadingDataDB}
          setBackupDB={setBackupDB}
          setStatusLoadingDataForApp={setLoadingDataAPP}
          setStatusLoadingDataForDB={setLoadingDataDB}
        />
      </NavigationContainer>
    )
    .toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});

test("Component BackupDB load DB complete return", async () => {
  let loadingDataAPP = 100;

  function setLoadingDataAPP(value: number) {
    loadingDataAPP = value;
  }

  let loadingDataDB = 100;

  function setLoadingDataDB(value: number) {
    loadingDataDB = value;
  }

  function setBackupDB(value: boolean) {
    return value;
  }

  const { getByTestId } = render(
    <NavigationContainer>
      <BackupDB
        loadingDataAPP={loadingDataAPP}
        loadingDataDB={loadingDataDB}
        setBackupDB={setBackupDB}
        setStatusLoadingDataForApp={setLoadingDataAPP}
        setStatusLoadingDataForDB={setLoadingDataDB}
      />
    </NavigationContainer>
  );

  fireEvent(getByTestId("closeAndReturn"), "onPress");
  expect(loadingDataAPP).toEqual(0);
});
