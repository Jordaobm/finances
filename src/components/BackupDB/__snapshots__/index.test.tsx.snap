// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomAutocomplete 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "rgba(1, 55, 148, 0.8)",
        "padding": 32,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M22 12C22 17.52 17.52 22 12 22C6.48 22 3.11 16.44 3.11 16.44M3.11 16.44H7.63M3.11 16.44V21.44M2 12C2 6.48 6.44 2 12 2C18.67 2 22 7.56 22 7.56M22 7.56V2.56M22 7.56H17.56"
          propList={
            Array [
              "stroke",
            ]
          }
          stroke={4294967295}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Bold",
          "fontSize": 20,
          "marginTop": 32,
        },
      ]
    }
  >
    Importando dados para o banco
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Regular",
          "fontSize": 12,
          "textAlign": "center",
        },
      ]
    }
  >
    Aguarde enquanto o finances tr√°s todos os seus dados do seu arquivo de backup
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Roboto-Regular",
          "fontSize": 12,
          "marginTop": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    10
     %
  </Text>
</View>
`;
